### PR TITLE
Milestone4

### PR DESCRIPTION
**Pull request by Ashah15**
Created the logic folder and initialize the calculate.js and operate.js file. Implemented the two files module and added the big.js dependency.

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    operate(total, next, operation);
+  } else if ((buttonName === '=') && (next && total)) {
+    total = operate(total, next, operation);
+    next = null;
+    operation = null;
+  } else if (buttonName === '.' && next) {
+    if (!next.includes('.')) {
+      next += '.';
+    }
+  }
```
Comment (uryelah) [inline src/logic/calculate.js]: - [ ] Please make sure that all possible cases are covered.
  - There are buttons without a conditional case, for example "+"
- [ ] What if there's only a single number entered? Make sure that value passes from next to total.
- [ ] Test your logic with console logs, passing mock objects to calculate, in order to debug the logic and make sure it works as a normal calculator for all the operations.
Ex:
```javaScript
console.log(calculate({total: 1, next: 4, operation: '+'}));
// what is the result of this? In theory it should be {total: 5, next: null, operation: null }
//what is your code returning instead?
```

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  if (operation === '+') {
+    result = firstNumber.plus(secondNumber);
+  } else if (operation === '-') {
+    result = firstNumber.minus(secondNumber);
+  } else if (operation === 'x') {
+    result = firstNumber.times(secondNumber);
+  } else if (operation === '÷') {
+    result = firstNumber.div(secondNumber);
+  } else if (operation === '%') {
```
Comment (uryelah) [inline src/logic/operate.js]: - [ ] The modifier button should turn the number into a percentage based number, e.g. 95 % == .95

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Asha** , I'm Sarah your reviewer for this PR 👩‍🚀 

Good job starting out this milestone! There are still logic issues to fix in the code so please make sure to test the calculate method yourself with console logs, compare it with the behavior of a real calculator, and then read my requests and make the necessary modifications.

Don't get discouraged though, you've done most of the job, just go on for a little more.

Read the points I mention below, work on the fixes, and submit for a new review.

Keep on coding and Rock on! 🎸 👨‍🎤 

# Require Changes ♻️ 

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Status: Invalid ❌ ❌ 

**Hello @Ashah15** 

### this pull request is marked as invalid for the following reasons
---

- [ ] Please fix all the linters issues before requesting another code review.
- [ ] you can fix the issue by returning a value directly inside the calculate (not wrapped in an if/else statement). 

---

please make these changes and request another code review

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @Ashah15**

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**

**Pull request merged**